MyApplication/app/src/main/java/com/joelimyx/myapplication/StockSyncAdapter.java

```diff
@@ -0,0 +1,77 @@
+package com.joelimyx.myapplication;
+
+import android.accounts.Account;
+import android.content.AbstractThreadedSyncAdapter;
+import android.content.ContentProvider;
+import android.content.ContentProviderClient;
+import android.content.ContentResolver;
+import android.content.ContentValues;
+import android.content.Context;
+import android.content.SyncResult;
+import android.database.Cursor;
+import android.os.Bundle;
+import android.util.Log;
+import android.widget.Toast;
+
+import com.android.volley.Request;
+import com.android.volley.RequestQueue;
+import com.android.volley.Response;
+import com.android.volley.VolleyError;
+import com.android.volley.toolbox.JsonObjectRequest;
+import com.android.volley.toolbox.Volley;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+
+/**
+ * Created by Joe on 12/8/16.
+ */
+
+public class StockSyncAdapter extends AbstractThreadedSyncAdapter {
+    private static final String TAG = "StockSyncAdapter";
+    private ContentResolver mContentResolver;
+    private Context mContext;
+
+    public StockSyncAdapter(Context context, boolean autoInitialize) {
+        super(context, autoInitialize);
+        mContext = context;
```

Comment:
Don't save a reference to context if you don't have to! I know you pass along the application context when you call this constructor, so you are not creating any memory leaks, but what if you were working in a team and another developer didn't understand and passed a service or activity reference?

In this case, all you need to save is the content resolver, not the context.